sprite.h

```diff
@@ -65,3 +65,11 @@ class Impostor : public Sprite {
 protected:
 	static BitmapName random_bitmap();
 };
+
+class Yoy : public Sprite {
```

Comment:
Architecture: I'm not a huge fan of making an entire new class for yoys here. The `Sprite` class is a base for customizing the **behavior** of the characters - i.e., the difference between a `Yonker` and an `Impostor` is one of behavior (Yonkers change texture and jitter around, Impostors do not.) The `Yoy` here is behaviorally identically to an `Impostor`.

That said, my suggestion is to put the yoys into the impostors. I see there's an intent here to make the yoys and impostors evenly distributed, so that there's a 50/50 chance of either - to keep that behavior I might prefer to modify the `Impostor` class to draw from two separate lists in `random_bitmap()`, with a 50/50 chance of selecting either list. Something like:

```c++
BitmapName Impostor::random_bitmap() {
	static std::vector<BitmapName> impostors = { cvjoy, nx, vx, lkmoyai, fn, fnplead };
	static std::vector<BitmapName> yoy = { lkyoy, lkyoyapprove, fnyoy, cvyoy };

        if (Noise::random() < 0.5f) {
	    return impostors[(int) (Noise::random() * impostors.size())];
        } else {
            return yoys[(int) (Noise::random() * yoys.size())];
        }
}
```

Reply:
Never thought about doing it like this. And imo it's way better. I've changed it.